src/Vector.cpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "Vector.h"
+
+
+Vector::Vector(const size_t& size, const int& value)
+    : _size(0), _capacity(0), _data(nullptr) {
+    for (size_t i = 0; i < size; i++) {
+        push_back(value);
+    }
+}
+
+Vector::~Vector() {
+    if (_data != nullptr)
+        delete[] _data;
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : _size(0), _capacity(0), _data(nullptr) {
+    *this = other;
+}
+
+Vector& Vector::operator=(const Vector& other) {
+    if (this == &other)
+        return *this;
+
+    for (int i = 0; i < other.size(); i++) {
+        push_back(other[i]);
+    }
+    return *this;
+}
+
+// Move Constructor
+Vector::Vector(Vector&& other) noexcept
+    : _size(0), _capacity(0), _data(nullptr)
+{
+    if (this != &other) {
+        swap(other);
+    }
+}
+
+// Move Assignment Operator
+Vector& Vector::operator=(Vector&& other) noexcept
+{
+    if (this != &other) {
+        swap(other);
+    }
+    return *this;
+}
+
+// Access operator
+int& Vector::operator[](int index) {
+    if (index < 0 || index >= _size) {
+        throw std::out_of_range("Index out of bounds");
+    }
+    return _data[index];
+}
+
+const int& Vector::operator[](int index) const {
+    if (index < 0 || index >= _size) {
+        throw std::out_of_range("Index out of bounds");
+    }
+    return _data[index];
+}
+
+int* Vector::data() const {
+    return _data;
+}
+
+bool Vector::empty() const {
+    return _size == 0;
+}
+
+size_t Vector::size() const {
+    return _size;
+}
+
+void Vector::reserve(const size_t& new_cap) {
+    if (new_cap > _capacity) {
+        int* new_data = new int[new_cap];
+        for (size_t i = 0; i < _size; i++) {
+            new_data[i] = _data[i];
+        }
+
+        if (_data != nullptr)
+            delete[] _data;
+        _data = new_data;
+        _capacity = new_cap;
+    }
+}
+
+size_t Vector::capacity() const {
+    return _capacity;
+}
+
+void Vector::clear() {
+    _size = 0;
+}
+
+void Vector::resize(const size_t& count, const int& value) {
+    if (count < _size) {
+        _size = count;
+    }
+    else if (count > _size) {
+        for (std::size_t i = _size; i < count; i++) {
```

Comment:
An essential part of readable code is uniformity. Please decide if you use `size_t` or `std::size_t` and stick to one